app/src/main/java/eu/zimbelstern/tournant/ui/MainActivity.kt

```diff
@@ -256,6 +257,18 @@ class MainActivity : AppCompatActivity(), RecipeListAdapter.RecipeListInterface
 			intent.action = ""
 		}
 
+		lifecycleScope.launch { // BACKUP RECIPES
+			viewModel.idsRecipesFiltered.collectLatest {
+				if (sharedPrefs.getBoolean(PREF_AUTO_BACKUP, false) && mode == MODE_STANDALONE && !sharedPrefs.getString(PREF_BACKUP_FILE, null).isNullOrEmpty()){
+					exportRecipes(it, "xml", false)
+					val uri = sharedPrefs.getString(PREF_BACKUP_FILE, "")?.toUri()
+					if (uri != null) {
+						viewModel.copyRecipesFromExportDir("export", "xml", uri)
```

Comment:
You are not handling a permission denial. The system will eventually revoke document access and this will trigger a SecurityException causing the app to crash. Also, the file may be deleted.